app/org/sagebionetworks/bridge/dynamodb/DynamoTaskDao.java

```diff
@@ -0,0 +1,226 @@
+        
+        finalizeTasks(tasks);
+        return tasks;
+    }
+    
+    /** {@inheritDoc} */
+    @Override
+    public List<Task> getTasksWithoutScheduling(User user) {
+        List<Task> tasks = Lists.newArrayList();
+        queryForTasks(user.getHealthCode(), tasks, null);
+        finalizeTasks(tasks);
+        return tasks;
+    }
+    
+    /** {@inheritDoc} */
+    @Override
+    public void updateTasks(String healthCode, List<Task> tasks) {
+        List<Task> tasksToSave = Lists.newArrayList();
+        for (Task task : tasks) {
+            task.setHealthCode(healthCode);
+            Task dbTask = mapper.load(task);
+            if (dbTask != null && (task.getStartedOn() != null || task.getFinishedOn() != null)) {
+                if (task.getStartedOn() != null) {
+                    dbTask.setStartedOn(task.getStartedOn());
+                }
+                if (task.getFinishedOn() != null) {
+                    dbTask.setFinishedOn(task.getFinishedOn());
+                }
+                tasksToSave.add(dbTask);
+            }
+        }
+        if (!tasksToSave.isEmpty()) {
+            List<FailedBatch> failures = mapper.batchSave(tasksToSave);
+            BridgeUtils.ifFailuresThrowException(failures);
+        }
+    }
+    
+    /** {@inheritDoc} */
+    @Override
+    public void deleteTasks(String healthCode) {
+        DynamoTask hashKey = new DynamoTask();
+        hashKey.setHealthCode(healthCode);
+
+        DynamoDBQueryExpression<DynamoTask> query = new DynamoDBQueryExpression<DynamoTask>().withHashKeyValues(hashKey);
+        
+        PaginatedQueryList<DynamoTask> queryResults = mapper.query(DynamoTask.class, query);
+        
+        // Confirmed that you have to transfer these tasks to a list or the batchDelete does not work. 
+        List<DynamoTask> tasksToDelete = Lists.newArrayListWithCapacity(queryResults.size());
+        tasksToDelete.addAll(queryResults);
+        
+        List<FailedBatch> failures = mapper.batchDelete(tasksToDelete);
+        BridgeUtils.ifFailuresThrowException(failures);
+    }
+
+    /**
+     * Filter out expired, deleted or finished tasks. We cannot do this in the query,
+     * or the scheduler will just add the tasks back again. We need to see they exist.
+     * @param tasks
+     */
+    private void finalizeTasks(List<Task> tasks) {
+        for (Iterator<Task> i = tasks.iterator(); i.hasNext();) {
+            if (HIDE_FROM_USER.contains(i.next().getStatus())) {
+                i.remove();
+            }
+        }
+        Collections.sort(tasks, TASK_COMPARATOR);
+    }
+
+    private void queryForTasks(String healthCode, List<Task> tasks, Set<String> guids) {
+        DynamoTask hashKey = new DynamoTask();
+        hashKey.setHealthCode(healthCode);
+        
+        // Why would you limit this? In terms of this query, get all the tasks.
+        DynamoDBQueryExpression<DynamoTask> query = new DynamoDBQueryExpression<DynamoTask>()
```

Comment:
- don't need a global secondary index here
- need all the tasks that are not expired or started (even if expired)
- either filter this server-side, or you'll have to filter on this side.
- but you cannot time limit this, as noted in comment